### PR TITLE
[REVIEW] Fix fetching of MultiIndex values when a label is passed

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #12259 

This PR fixes an issue with `.loc` where a raw string can be passed like `'a'`(errors on `branch-23.02`), instead of `('a', )`(which works on `branch-23.02`).

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/multiindex.py [hunk 1]
```diff
@@ -859,7 +859,9 @@ def _get_row_major(
             list(row_tuple) if isinstance(row_tuple, tuple) else row_tuple
         ):
             return df[row_tuple]
-        if isinstance(row_tuple, slice):
+        if isinstance(row_tuple, str):
```
Comment (mroeschke) [inline python/cudf/cudf/core/multiindex.py:862]: Hmm wouldn't this be an issue generally if a scalar is passed? e.g.

```
df = pd.DataFrame([[1],[2]], index=[[1, 1],[2,3]])
df.loc[1]
```

Comment (galipremsagar) [reply python/cudf/cudf/core/multiindex.py:862]: That already works since take below can handle integer like inputs(`1`)

Comment (mroeschke) [reply python/cudf/cudf/core/multiindex.py:862]: If it understand `_get_valid_indices_by_tuple` and then `take`, these will treat integers as positional when in `loc` we want to treat integers as labels?

Comment (galipremsagar) [reply python/cudf/cudf/core/multiindex.py:862]: Yeah, that's what is happening incase of MultiIndex for both pandas & cudf(matching pandas)

Comment (mroeschke) [reply python/cudf/cudf/core/multiindex.py:862]: On pandas 1.5 integers are treated as labels when using `loc` with a MultiIndex

```
>>> import pandas as pd
pd.__version__>>> pd.__version__
'1.5.2'
>>> df = pd.DataFrame([[1],[2]], index=[[-1, -2], [2, 3]])
>>> df
      0
-1 2  1
-2 3  2
>>> df.loc[-1]
   0
2  1
>>> df.loc[1]
KeyError: 1
```

Comment (galipremsagar) [reply python/cudf/cudf/core/multiindex.py]: Thanks @mroeschke, I've discovered that the above example doesn't work correctly in `cudf` and fixed it in this PR properly. Added parameters for the same test coverage. This should be ready for review.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**